content/docs/guides/continuous-delivery/gitlab-ci.md

```diff
@@ -193,10 +207,249 @@ pulumi config set mysetting myvalue
+        url: https://app.pulumi.com/acme/product-catalog/${CI_COMMIT_REF_SLUG}
+        auto_stop_in: 2 days
+        on_stop: ephem-env:tear-down
+    only:
+        - merge_requests
+
+# Specialized tear down script, so you can manually cleanup any environment.
+# Will also be triggered if the ephemeral environment expires.
+ephem-env:tear-down:
+    stage: deploy
+    # Necessary, since the branch may have been deleted.
+    variables:
+        GIT_STRATEGY: none
+    script:
+        - echo "=== Tearing Down Ephemeral Environment ==="
+        - sh ./scripts/run-pulumi.sh destroy  ${CI_COMMIT_REF_SLUG}
+        - sh ./scripts/run-pulumi.sh stack-rm ${CI_COMMIT_REF_SLUG}
+    environment:
+        name: Pulumi stack ${CI_COMMIT_REF_SLUG}
+        action: stop
+    when: manual
+```
+
+And that's it! There are a few more important details to be aware of, but by
+simply adding these two pipeline jobs in enough to start taking advantage of
+ephemeral environments!
+
+### Detailing `run-pulumi.sh`
+
+Like before, we need to manage the interactions with the Pulumi CLI. And to do that we rely on a
+script called `run-pulumi.sh`. However, a couple of changes are made to support ephemeral environments.
+
+```bash
+#!/bin/bash
+#
+# Invokes Pulumi to create or update a stack.
+
+# Installs the Pulumi CLI. Essentially
+# $ curl -fsSL https://get.pulumi.com/ | bash -s
+. ./scripts/install-pulumi.sh
+
+# Confirm pulumi is setup correctly.
+pulumi login
+pulumi whoami --verbose
+
+readonly PULUMI_OPERATION=${1}
+readonly STACK_NAME=${2}
+
+# CD to Pulumi project folder.
+cd infrastructure
+
+pulumi stack select ${STACK_NAME}
+if [ $? -ne 0 ]; then
+    echo "Stack '${STACK_NAME}' does not exist. Creating..."
+    pulumi stack init ${STACK_NAME}
+
+    # When creating ephemeral environments, we need to use some default
+    # configuration values.
+    pulumi config set machineSize t2.small
+    pulumi config set rootDomain testing.acme-corp.net
+fi
+
+# For ephemeral environments, we need to pull down config since it isn't checked
+# into the repository.
+if [ "${STACK_NAME}" != "production" ]; then
+    echo "Downloading configuration data for ephemeral environment..."
+    pulumi config refresh
+    if [ $? -ne 0 ]; then
+        echo "Error pulling previous config. Ignoring, since probably just first-run."
+    fi
+fi
+
+case "${PULUMI_OPERATION}" in
+    "update")
+        pulumi up --yes
+        ;;
+    "preview")
+        pulumi preview
+        ;;
+    "destroy")
+        pulumi destroy --yes
+        ;;
+    "stack-rm")
+        pulumi stack rm --yes
+        ;;
+    *)
+        echo "Error: Unrecognized operation '${PULUMI_OPERATION}'"
+        exit 1
+        ;;
+esac
+
+echo "Done"
+```
+
+#### Stack Initializatiom
+
+There isn't an easy way to determine if we are updating the ephemeral environment for the first or
+the 100th time. So in the script, we attempt to run `pulumi stack select`, and check `pulumi`'s exit code
```

Comment:
Just wondering if this could be skipped and just pass `--create/-c` to `pulumi stack select`? It would simplify this.